src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs

```diff
@@ -2059,13 +2107,26 @@ internal override IList<PSModuleInfo> ImportModulesUsingWinCompat(IEnumerable<st
                 }
             }
 
+            var winPSVersionString = Utils.GetWindowsPowerShellVersionFromRegistry();
+            if (!winPSVersionString.StartsWith("5.1", StringComparison.OrdinalIgnoreCase))
+            {
+                string errorMessage = string.Format(CultureInfo.InvariantCulture, Modules.WinCompatRequredVersionError, winPSVersionString);
+                throw new InvalidOperationException(errorMessage);
+            }
```

Comment:
Could ou please explain why do we the check after  PrepareNoClobberWinCompatModuleImport call?

Reply:
I don't see the confusion. It's still preparation, but just it may fail and stop the remote session creation.

Reply:
_It was a preparation_ while we fast return but now, at first, we are trying to do _regular loading_ and only then _fallback_ to wincompat loading.

Reply:
> but now, at first, we are trying to do regular loading and only then fallback to wincompat loading.

No, it has always been working in this way. Take a look at the existing code here https://github.com/PowerShell/PowerShell/blob/2e8009bf721f3997bbcf710a8c683f94778f97c5/src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs#L1984-L2015